app/serializers/v3/product_serializer.rb

```diff
@@ -11,11 +11,12 @@ def repositories
   end
 
   attributes :id, :name, :identifier, :former_identifier, :version, :release_type, :arch,
-             :friendly_name, :product_class, :cpe, :free, :description, :eula_url, :repositories, :product_type, :extensions
+             :friendly_name, :product_class, :cpe, :free, :description, :eula_url, :repositories, :product_type,
+             :extensions, :recommended, :available
 
   def extensions
-    object.mirrored_extensions.map do |extension|
-      ::V3::ProductSerializer.new(extension, base_url: base_url).attributes
+    object.extensions.for_root_product(root_product).map do |extension|
+      ::V3::ProductSerializer.new(extension, base_url: base_url, root_product: root_product).attributes
     end
```

Comment:
In [glue](https://github.com/SUSE/happy-customer/blob/master/glue/app/serializers/v3/product_serializer.rb#L5) we simply have a `has_many :extensions`, and all the `instance_options` are passed down to the child serializers automatically, so you don't need to specify `base_url` or `root_product`.
That leaves you with:

```ruby
has_many :extensions, serializer: V3::ProductSerializer

def extensions
  object.extensions.for_root_product(root_product)
end
```

Reply:
I remember I was looking into this at the very beginning and the current code is the solution that works correctly.

The problem is that for some reason it doesn't serialize all of the attributes of the relation, namely the ones that are declared on the serializer and don't exist on the model (like `available` and `recommended`). With the `has_many` relation they seem to be just missing.

If you have any idea what's going on with that -- feel free to share.